sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jms/ServiceBusJmsConnectionFactoryConfiguration.java

```diff
@@ -17,28 +24,70 @@
 import org.springframework.boot.autoconfigure.jms.JmsProperties;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.context.support.GenericApplicationContext;
 import org.springframework.jms.connection.CachingConnectionFactory;
+import org.springframework.util.StringUtils;
 
+import java.util.Properties;
 import java.util.stream.Collectors;
 
+import static com.azure.spring.cloud.service.implementation.identity.credential.provider.SpringTokenCredentialProvider.PASSWORDLESS_TOKEN_CREDENTIAL_BEAN_NAME;
+
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnMissingBean(ConnectionFactory.class)
-class ServiceBusJmsConnectionFactoryConfiguration {
+class ServiceBusJmsConnectionFactoryConfiguration  {
+
+    private static final Log LOGGER = LogFactory.getLog(ServiceBusJmsConnectionFactoryConfiguration.class);
+    private final GenericApplicationContext applicationContext;
+
+    ServiceBusJmsConnectionFactoryConfiguration(GenericApplicationContext applicationContext) {
+        this.applicationContext = applicationContext;
+    }
 
-    private static ServiceBusJmsConnectionFactory createJmsConnectionFactory(AzureServiceBusJmsProperties properties,
+    private ServiceBusJmsConnectionFactory createJmsConnectionFactory(AzureServiceBusJmsProperties jmsProperties,
                                                                              ObjectProvider<AzureServiceBusJmsConnectionFactoryCustomizer> factoryCustomizers) {
-        return new ServiceBusJmsConnectionFactoryFactory(properties,
+        TokenCredentialProvider tokenCredentialProvider = getPasswordlessTokenCredentialProvider(jmsProperties);
+        return new ServiceBusJmsConnectionFactoryFactory(tokenCredentialProvider, jmsProperties,
             factoryCustomizers.orderedStream().collect(Collectors.toList()))
             .createConnectionFactory(ServiceBusJmsConnectionFactory.class);
     }
 
+    private TokenCredentialProvider getPasswordlessTokenCredentialProvider(AzureServiceBusJmsProperties serviceBusJmsProperties) {
```

Comment:
can we leverage the authenticate template instead?

Reply:
We should put all the same logic into the same place, instead of writing it everywhere

Reply:
Will improve the logic